### DISCUSSION TITLE
Is there a general function for making the engine progress any pending request?

### DISCUSSION THREAD

**Discussion by miEsMar** [pmodels/mpich#7129] [General]
Hi everyone,

I was wondering if there is a general function that can be called using MPICH MPI Implementation to explicitly ask the underlying engine to progress any incomplete request that might exist.
Preferably a layer above the actual engine (CH3 or CH4) interface, so to avoid the use of macros to specialize actual calls.
Similar to `opal_progress()` for OpenMPI.

Thanks.

**Comment by raffenet** [comment]
@miEsMar if you are using MPICH >= 4.1.0, you can use `MPIX_Stream_progress(MPIX_STREAM_NULL);`

**Reply by miEsMar** [reply]
Thanks for the valuable information @raffenet !
For previous versions what would it be the preferred approach?
To better navigate the code I cloned the `main` branch, and there I see there's the `MPID_Stream_progress()` function, not `MPIX`.
Can I use that?
PS: I see that in case of using MPICH3, it is actually defined as a macro redirecting to `MPIDI_CH3_Progress_poke()`.|
**EDIT**: I ask since on our systems it is highly the case that the 3.x version is provided for some machines, and I would love the tool to work regardless of the actual engine used.
